### ISSUE TITLE
ensure_proper_audience not used

### ISSUE THREAD

**Issue by paneq** [basecamp/google_sign_in#3]
Hey, based on the code below (extracted from the repository) I believe that `ensure_proper_audience` does not serve any purpose.

```ruby
module GoogleSignIn
  class Identity
    def initialize(token)
      ensure_client_id_present
      set_extracted_payload(token)
      ensure_proper_audience
    end

   private
    
      def ensure_proper_audience
        @payload["aud"].include?(client_id)
      end
```

It does not set any instance variable nor raises an exception. Why do we need to check it and what should happen if it returns `false` ?

Similarly named `ensure_client_id_present` raises an exception.

**Comment by dhh** [comment]
Yes, it needs to raise an exception. Please do prep a PR with a test, if you feel like it!

**Comment by paneq** [comment]
@dhh Sure thing. Thank you for the clarification.

**Comment by dhh** [comment]
I actually just pushed a fix for this. But we need to have a testing
structure setup to validate it. Please do look into that if you'd like.

On Fri, Jul 28, 2017 at 9:43 AM, Robert Pankowecki <notifications@github.com

**Comment by paneq** [comment]
@dhh :+1: 

Check out #5 for tests. Could you release a new version with the security fix?

**Comment by dhh** [comment]
Released 0.1.1 with the fix.
